src/commands/test.js

```diff
@@ -2,70 +2,134 @@ const { Binary } = require('binary-install-raw')
 const os = require('os')
 const chalk = require('chalk')
 const fetch = require('node-fetch')
+const { filesystem, print } = require('gluegun')
+const { fixParameters } = require('../command-helpers/gluegun')
 const semver = require('semver')
+const { spawn, exec } = require('child_process')
+const yaml = require('js-yaml')
 
 const HELP = `
 ${chalk.bold('graph test')} ${chalk.dim('[options]')} ${chalk.bold('<datasource>')}
 
 ${chalk.dim('Options:')}
-
-  -f  --force                   Overwrite folder + file when downloading
+  -c, --coverage                Run the tests in coverage mode. Works with v0.2.1 and above
+  -d, --docker                  Run the tests in a docker container(Note: Please execute from the root folder of the subgraph)
+  -f  --force                   Binary - overwrites folder + file when downloading. Docker - rebuilds the docker image
   -h, --help                    Show usage information
   -l, --logs                    Logs to the console information about the OS, CPU model and download url (debugging purposes)
+  -r, --recompile               Force-recompile tests (Not available in 0.2.2 and earlier versions)
   -v, --version <tag>           Choose the version of the rust binary that you want to be downloaded/used
   `
 
 module.exports = {
   description: 'Runs rust binary for subgraph testing',
   run: async toolbox => {
-    // Obtain tools
-    let { print } = toolbox
-
     // Read CLI parameters
-    let { f, force, h, help, l, logs, v, version } = toolbox.parameters.options
-    let datasource = toolbox.parameters.first
+    let {
+      c,
+      coverage,
+      d,
+      docker,
+      f,
+      force,
+      h,
+      help,
+      l,
+      logs,
+      r,
+      recompile,
+      v,
+      version,
+    } = toolbox.parameters.options
 
+    let opts = new Map()
     // Support both long and short option variants
-    force = force || f
-    help = help || h
-    logs = logs || l
-    version = version || v
+    opts.set("coverage", coverage || c)
+    opts.set("docker", docker || d)
+    opts.set("force", force || f)
+    opts.set("help", help || h)
+    opts.set("logs", logs || l)
+    opts.set("recompile", recompile || r)
+    opts.set("version", version || v)
+
+    // Fix if a boolean flag (e.g -c, --coverage) has an argument
+    try {
+      fixParameters(toolbox.parameters, {
```

Comment:
I don't think this is doing anything if you're not reading the result of `fixParameters` or the `toolbox.parameters` itself.

Reply:
I thought so too, but I console logged it and it seems it updates `toolbox.parameters.array`, although I'm not sure why
<img width="453" alt="Screenshot 2021-12-23 at 17 02 02" src="https://user-images.githubusercontent.com/20456492/147258110-6f9450e7-a802-4eb3-8f52-8bd7834613fb.png">

Reply:
And I'm assigning the datasource here after `fixParameters` from `toolbox.parameters.array[0]` https://github.com/graphprotocol/graph-cli/pull/796/files/26f58501b72dab3e03f0768ec1918244a2db7be4#diff-7b8030c07614a0f2a6ec365e75d5bacf2fd305916553c879d3f5e2832a5b643eR77